activities.el

```diff
@@ -434,13 +434,14 @@ With PERSISTP, persist to disk (otherwise see
 `activities-always-persist').  To be safe for `kill-emacs-hook',
 this demotes errors."
   (interactive)
-  (with-demoted-errors "activities-save-all: ERROR: %S"
-    (dolist (activity (cl-remove-if-not #'activities-activity-active-p (map-values activities-activities)))
-      (let ((activities-saving-p t)
-            ;; Don't write to disk for each activity.
-            (activities-always-persist nil))
-        (activities-save activity :lastp t)))
-    (activities--persist persistp)))
+  (unless (run-hook-with-args-until-success 'activities-anti-save-predicates)
```

Comment:
My idea was to have a new, global anti-save-predicates list that would prevent saving of any activities.  I think that the existing anti-save-predicates list should continue to exist as a per-activity check done after switching to an activity, before saving it; making that list global now would be a change in functionality that could break user configuration.  Sorry if I wasn't clear.

Reply:
> My idea was to have a new, global anti-save-predicates list ...

Relaying what I think you mean:

- `activities-anti-save-predicates` is to be "activity" specific, meaning if there is an activity _**Foo**_ then there could be an `activities-anti-save-predicates` that would only apply (_be local_) to activity _**Foo**_.

- Add an `activities-global-anti-save-predicates` (or similarly named) variable that, given having 2 activities _**Foo**_ and _**Bar**_, would apply to both activities and prevent saving of both.

Am I on the right track for what you were thinking?  Based your response, I will probably have a few follow-up questions.

> making that list global now would be a change in functionality that could break user configuration

You do caveat in the README: **_Because `activities` is at an early stage of development..._** :)

Reply:
What I mean is that:

1. `activities-anti-save-predicates` should be unchanged from how it works now.  When about to save an activity, after switching to it, and before saving it, these predicates are called to see whether the activity should be saved.
2. A new, similar list of predicates should be defined for globally preventing *any* saving of activities.  Its predicates should be called before switching to any activities, and if any of them return non-nil, there should be no attempt to save any activity, nor should any activity be switched to.